pkg/controller/core/workload_controller.go

```diff
@@ -216,6 +221,25 @@ func (r *WorkloadReconciler) reconcileSyncAdmissionChecks(ctx context.Context, w
 	return false, nil
 }
 
+func (r *WorkloadReconciler) reconcileReservationOnQueueDeletion(ctx context.Context, wl *kueue.Workload, cqName string) (bool, error) {
+	if workload.IsAdmitted(wl) {
+		return false, nil
+	}
+
+	queue := kueue.ClusterQueue{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: cqName}, &queue); err != nil {
```

Comment:
just in case, also remove the reservation if this returns a NotFound error

Reply:
done